File: system_code/Customer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package call_center_simulation;

import java.util.Date;

/**
 *
 * @author Jack Napier
 */
public class Customer {
    private int callId;
    private Date arrivalTime;
    private int interArrivalTime;
    private int serviceLength;
    private int waitingTime1 = 0;
    private int servingTime1 = 0;
    private int waitingTime2 = 0;
    private int servingTime2 = 0;


    public int getCallId() {
        return callId;
    }

    public void setCallId(int callId) {
        this.callId = callId;
    }

    public Date getArrivalTime() {
        return arrivalTime;
    }

    public void setArrivalTime(Date arrivalTime) {
        this.arrivalTime = arrivalTime;
    }
    

    public int getInterArrivalTime() {
        return interArrivalTime;
    }

    public void setInterArrivalTime(int interArrivalTime) {
        this.interArrivalTime = interArrivalTime;
    }

    public int getServiceLength() {
        return serviceLength;
    }

    public void setServiceLength(int serviceLength) {
        this.serviceLength = serviceLength;
    }
    
    public int getWaitingTime1() {
        return waitingTime1;
    }

    public void setWaitingTime1(int waitingTime1) {
        this.waitingTime1 = waitingTime1;
    }

    public int getServingTime1() {
        return servingTime1;
    }

    public void setServingTime1(int servingTime1) {
        this.servingTime1 = servingTime1;
    }

    public int getWaitingTime2() {
        return waitingTime2;
    }

    public void setWaitingTime2(int waitingTime2) {
        this.waitingTime2 = waitingTime2;
    }

    public int getServingTime2() {
        return servingTime2;
    }

    public void setServingTime2(int servingTime2) {
        this.servingTime2 = servingTime2;
    }
}

File: main_code/Customer.py
from datetime import date,time
class Customer:
    def __init__(self):
        self.__servingTime2 = 0
        self.__waiting_time2 = 0
        self.__serving_time1 = 0
        self.__waiting_time1 = 0
        self.__service_length = 0
        self.__inter_arrival_time = 0
        self.__arrivalTime = date.today
        self.__callId =0

    def get_call_id(self):
        return self.__callId

    def set_call_id(self, call_id):
        self.__callId = call_id

    def get_arrival_time(self):
        return self.__arrivalTime

    def set_arrival_time(self,arrival_time):
        self.__arrivalTime = arrival_time

    def get_inter_arrival_time(self):
        return self.__inter_arrival_time

    def set_inter_arrival_time(self,inter_arrival_time):
        self.__inter_arrival_time = inter_arrival_time

    # def get_service_length(self):
    #     return self.__service_length

    # def set_service_length(self, service_length):
    #     self.__service_length = service_length

    def get_waiting_time(self):
        return self.waiting_time1

    def set_waiting_time(self, waiting_time1):
        self.__waiting_time1 = waiting_time1

    # def get_serving_time1(self):
    #     return self.__serving_time1

    # def set_serving_time1(self, serving_time1):
    #     self.__serving_time1 = serving_time1

    # def get_waiting_time2(self):
    #     return self.__waiting_time2

    # def set_waiting_time2(self, waiting_time2):
    #     self.__waiting_time2 = waiting_time2;

    # def get_serving_time2(self):
    #     return self.__serving_time2;

    # def set_serving_time2(self, serving_time2):
    #     self.__servingTime2 = serving_time2

